website/addons/gdrive/model.py

```diff
@@ -0,0 +1,280 @@
+                    'download': url + '?action=download'
+                },
+            },
+        )
+
+    def find_or_create_file_guid(self, path):
+        return AddonGdriveGuidFile.get_or_create(self.owner, path)
+
+    # #### Callback overrides #####
+
+    def before_register_message(self, node, user):
+        """Return warning text to display if user auth will be copied to a
+        registration.
+        """
+        category, title = node.project_or_component, node.title
+        if self.user_settings and self.user_settings.has_auth:
+            return (
+                u'The contents of Google Drive add-ons cannot be registered at this time; '
+                u'the Google Drive folder linked to this {category} will not be included '
+                u'as part of this registration.'
+            ).format(**locals())
+
+    # backwards compatibility
+    before_register = before_register_message
+
+    def before_fork_message(self, node, user):
+        """Return warning text to display if user auth will be copied to a
+        fork.
+        """
+        category = node.project_or_component
+        if self.user_settings and self.user_settings.owner == user:
+            return (u'Because you have authorized the Google Drive add-on for this '
+                    '{category}, forking it will also transfer your authentication token to '
+                    'the forked {category}.').format(category=category)
+
+        else:
+            return (u'Because the Google Drive add-on has been authorized by a different '
+                    'user, forking it will not transfer authentication token to the forked '
+                    '{category}.').format(category=category)
+
+    # backwards compatibility
+    before_fork = before_fork_message
+
+    def before_remove_contributor_message(self, node, removed):
+        """Return warning text to display if removed contributor is the user
+        who authorized the Gdrive addon
+        """
+        if self.user_settings and self.user_settings.owner == removed:
+            category = node.project_or_component
+            name = removed.fullname
+            return (u'The Google Drive add-on for this {category} is authenticated by {name}. '
+                    'Removing this user will also remove write access to Dropbox '
+                    'unless another contributor re-authenticates the add-on.'
+                    ).format(**locals())
+
+    # backwards compatibility
+    before_remove_contributor = before_remove_contributor_message
+
+    def after_register(self, node, registration, user, save=True):
+        """After registering a node, copy the user settings and save the
+        chosen folder.
+
+        :return: A tuple of the form (cloned_settings, message)
+        """
+        clone, message = super(AddonGdriveNodeSettings, self).after_register(
+            node, registration, user, save=False
+        )
+        # Copy user_settings and add registration data
```

Comment:
In fact, there should be an automated test to verify that node addon settings do *not* get copied to registrations. See here for an example: https://github.com/kushG/osf.io/blob/5b47e8b9a3367c2d9f7b1d43c5aba9df00cbec9d/website/addons/dropbox/tests/test_models.py#L302